src/main/java/de/tum/in/www1/artemis/service/StudentExamService.java

```diff
@@ -248,6 +252,43 @@ else if (exercise instanceof ModelingExercise) {
         }
     }
 
+    /**
+     * Automatically assess the modeling- and programming exercises of student exams of an exam which are not submitted with 0 points.
+     * The assessment is counted as {@link AssessmentType#SEMI_AUTOMATIC} to make sure it is not considered for manual assessment,
+     * see {@link StudentParticipationRepository#findByExerciseIdWithLatestSubmissionWithoutManualResultsAndNoTestRunParticipation}.
+     * The assessor is current instructor who made the request.
+     *
+     * @param examId the exam id
+     * @return returns the number of assessedSubmissions
+     */
+    public int automaticallyAssessUnsubmittedExams(final Long examId) {
+        int numberOfAssessedSubmissions = 0;
+        User instructor = userService.getUser();
+        Set<StudentExam> unsubmittedStudentExams = findAllUnsubmittedStudentExams(examId);
+        Map<User, List<Exercise>> exercisesOfUser = unsubmittedStudentExams.stream().collect(Collectors.toMap(StudentExam::getUser, studentExam -> studentExam.getExercises()
+                .stream().filter(exercise -> exercise instanceof ModelingExercise || exercise instanceof TextExercise).collect(Collectors.toList())));
```

Comment:
Should we rather add an `abstract` method to `Exercise` or something that returns whether an exercise needs manual assessment? I think we keep adding these is Modeling/Text(/File) checks all over the place and we are actually interested in the assessment type.

Reply:
@jpbernius Thats a good point, but should we then include programming exercises configured with `SEMI_AUTOMIC` assessment type? In this feature for example we don't grade those, and there are other parts in the code where we specifically target modeling and text exercises.

Reply:
Ok, I understand. Wouldn't we want to exclude programming submissions as well in this situation. So they might be relevant as well? (I guess out of scope here though.)
We can keep this as is if you want.